Hide ssl deprecated constants

mpg opened the pull request:
This is a follow-up to #4495 and fixes #4285 (as the remaining parts are already tracked elsewhere). It removes the constants related to TLS 1.0 and TLS 1.1 from the public API and keeps them in a private header, as some code is still using them until we execute #4564.

Edit: turns out it was inconvenient to do that while `mbedtls_ssl_conf_ciphersuites_for_version()` was still part of the public API . It was already being removed in #4442, so I cherry-picked and completed #4442 (after discussing with Hanno) as that seem more convenient that having a dependency between two small PRs.

daverodgman reviewed (dismissed):
This seems good. Confirmed that the files affected by moving the version numbers to ssl_misc.h build OK.

gilles-peskine-arm reviewed (changes requested):
Needs a changelog entry and a migration guide entry since it's removing public API functions. Other than that LGTM.

mpg commented:
I've updated the existing ChangeLog and migration guide entries for "Remove 1.0 and 1.1" because this is all related. I've taken the opportunity to slightly improve the wording.

Note: this PR also removes `mbedtls_ssl_get_protocol_version_ciphersuites()` but that function was never part of a released version - it was introduced when we removed support for SSL 3.0. This is why I'm not mentioning its removal in the ChangeLog or migration guide.

gilles-peskine-arm commented on docs/3.0-migration-guide.d/remove_support_for_tls_1.0_1.1_and_dtls_1.0.md:
```diff
@@ -3,9 +3,25 @@ Remove suport for TLS 1.0, 1.1 and DTLS 1.0
+
+As a consequence of removing 1.0, support for CBC record splitting was also
+removed, as it was a work-around for a weakness in this particular version.
+There is no migration path is no longer makes sense with newer versions.
```
```suggestion
There is no migration path as it no longer makes sense with newer versions.
```
or
```suggestion
There is no migration path since the feature is no longer relevant.
```

gilles-peskine-arm commented on docs/3.0-migration-guide.d/remove_support_for_tls_1.0_1.1_and_dtls_1.0.md:
```diff
@@ -3,9 +3,25 @@ Remove suport for TLS 1.0, 1.1 and DTLS 1.0
 
 The migration path is to adopt the latest versions of the protocol.
+
+As a consequence of removing 1.0, support for CBC record splitting was also
```
```suggestion
As a consequence of removing (D)TLS 1.0, support for CBC record splitting was also
```

mpg replied:
I made that "TLS 1.0" since DTLS 1.0 never needed record splitting either, as it's based on TLS 1.1...

gilles-peskine-arm commented on ChangeLog.d/issue4286.txt:
```diff
@@ -1,11 +1,14 @@
+   * Remove support for TLS 1.0, TLS 1.1 and DTLS 1.0, as well as support for
+     CBC record splitting, fallback SCSV, and the ability to configure
+     ciphersuites per version, which are no longer relevant. This removes the
+     following public constants: MBEDTLS_SSL_PROTO_TLS1,
```
Non-blocker: I don't think we need to list every single constant that was removed. The description of the removed features is enough.

mpg replied:
I kept only the config options then. Could be useful for someone adapting their config scripts or custom file.

gilles-peskine-arm reviewed (changes requested):
A little copyediting

daverodgman reviewed (approved):
Updates are a good improvement to the Changelog.
